tmt/steps/execute/__init__.py

```diff
@@ -227,36 +227,34 @@ def check_shell(self, test: "tmt.Test") -> List["tmt.Result"]:
         assert test.returncode is not None
         assert test.real_duration is not None
         # Prepare the log path
-        data = {'log': self.data_path(test, TEST_OUTPUT_FILENAME),
-                'duration': test.real_duration}
+        data = ResultData(log=self.data_path(test, TEST_OUTPUT_FILENAME),
+                          duration=test.real_duration)
         # Process the exit code
         try:
-            data['result'] = {0: 'pass', 1: 'fail'}[test.returncode]
+            data.result = {0: ResultOutcome.PASS, 1: ResultOutcome.FAIL}[test.returncode]
         except KeyError:
-            data['result'] = 'error'
+            data.result = 'error'
             # Add note about the exceeded duration
             if test.returncode == tmt.utils.PROCESS_TIMEOUT:
-                data['note'] = 'timeout'
+                data.note = 'timeout'
                 self.timeout_hint(test)
         return [tmt.Result(data, test=test)]
 
     def check_beakerlib(self, test: "tmt.Test") -> List["tmt.Result"]:
         """ Check result of a beakerlib test """
         # Initialize data, prepare log paths
-        data: tmt.result.ResultData = {'result': 'error',
-                                       'log': [],
-                                       'duration': test.real_duration}
+        data = ResultData(result='error', log=[], duration=test.real_duration)
```

Comment:
`log` can be left out, `result` needs to be of correct type, i.e. `ResultOutcome`.